cores/esp32/chip-debug-report.cpp

```diff
@@ -0,0 +1,338 @@
+    uint32_t f_freq = 0;
+    switch(fhdr.spi_speed) {
+      case 0x0: f_freq = 40; break;
+      case 0x1: f_freq = 26; break;
+      case 0x2: f_freq = 20; break;
+      case 0xf: f_freq = 80; break;
+      default: f_freq = fhdr.spi_speed; break;
+    }
+    log_printf("  Bus Speed         : %lu MHz\n", f_freq);
+  }
+  log_printf("  Bus Mode          : ");
+#if CONFIG_ESPTOOLPY_OCT_FLASH
+  log_printf("OPI\n");
+#elif CONFIG_ESPTOOLPY_FLASHMODE_QIO
+  log_printf("QIO\n");
+#elif CONFIG_ESPTOOLPY_FLASHMODE_QOUT
+  log_printf("QOUT\n");
+#elif CONFIG_ESPTOOLPY_FLASHMODE_DIO
+  log_printf("DIO\n");
+#elif CONFIG_ESPTOOLPY_FLASHMODE_DOUT
+  log_printf("DOUT\n");
+#endif 
+}
+
+static void printPartitionsInfo(void){
+  log_printf("Partitions Info:\n");
+  log_printf("------------------------------------------\n");
+  esp_partition_iterator_t iterator = esp_partition_find(ESP_PARTITION_TYPE_ANY, ESP_PARTITION_SUBTYPE_ANY, NULL);
+  if(iterator != NULL){
+    esp_partition_iterator_t it = iterator;
+    while(it != NULL){
+      const esp_partition_t* partition = esp_partition_get(it);
+      if(partition){
+        log_printf("  %17s : addr: 0x%08X, size: %7.1f KB", partition->label, partition->address, b2kb(partition->size));
+        if(partition->type == ESP_PARTITION_TYPE_APP){
+          log_printf(", type:  APP");
+          if(partition->subtype == 0){
+            log_printf(", subtype: FACTORY");
+          } else if(partition->subtype >= 0x10 && partition->subtype < 0x20){
+            log_printf(", subtype: OTA_%lu", partition->subtype - 0x10);
+          } else if(partition->subtype == 0x20){
+            log_printf(", subtype: TEST");
+          } else {
+            log_printf(", subtype: 0x%02X", partition->subtype);
+          }
+        } else {
+          log_printf(", type: DATA");
+          log_printf(", subtype: ");
+          switch(partition->subtype){
+            case ESP_PARTITION_SUBTYPE_DATA_OTA: log_printf("OTA"); break;
+            case ESP_PARTITION_SUBTYPE_DATA_PHY: log_printf("PHY"); break;
+            case ESP_PARTITION_SUBTYPE_DATA_NVS: log_printf("NVS"); break;
+            case ESP_PARTITION_SUBTYPE_DATA_COREDUMP: log_printf("COREDUMP"); break;
+            case ESP_PARTITION_SUBTYPE_DATA_NVS_KEYS: log_printf("NVS_KEYS"); break;
+            case ESP_PARTITION_SUBTYPE_DATA_EFUSE_EM: log_printf("EFUSE_EM"); break;
+            case ESP_PARTITION_SUBTYPE_DATA_UNDEFINED: log_printf("UNDEFINED"); break;
+            case ESP_PARTITION_SUBTYPE_DATA_ESPHTTPD: log_printf("ESPHTTPD"); break;
+            case ESP_PARTITION_SUBTYPE_DATA_FAT: log_printf("FAT"); break;
+            case ESP_PARTITION_SUBTYPE_DATA_SPIFFS: log_printf("SPIFFS"); break;
+            default: log_printf("0x%02X", partition->subtype); break;
+          }
+        }
+        log_printf("\n");
+      }
+      it = esp_partition_next(it);
+    }
+    //esp_partition_iterator_release(iterator);
+  }
+}
+
+static void printSoftwareInfo(void){
```

Comment:
I think it is worth printing MAC addresses for BLE, WiFi, and Ethernet when it is available.

Reply:
why? how will that help us?

Reply:
It's just because sometimes people don't know how to print the MAC or get confused with the 3 MACs.
If you think this info is useless for us then it's ok.

Reply:
I would rather not collect such unique information and I do not think that it will help us troubleshooting anything. If users can't get MAC addresses themselves, we can help separately.